csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/Patterns/PositionalPattern.cs

```diff
@@ -9,11 +9,7 @@ internal class PositionalPattern : Expression
         internal PositionalPattern(Context cx, PositionalPatternClauseSyntax posPc, IExpressionParentEntity parent, int child) :
             base(new ExpressionInfo(cx, null, cx.CreateLocation(posPc.GetLocation()), ExprKind.POSITIONAL_PATTERN, parent, child, false, null))
         {
-            child = 0;
-            foreach (var sub in posPc.Subpatterns)
-            {
-                Expressions.Pattern.Create(cx, sub.Pattern, this, child++);
-            }
+            posPc.Subpatterns.ForEach((p, i) => Pattern.Create(cx, p.Pattern, this, i));
```

Comment:
There might be some more occurrences of this pattern in our codebase. Searching for the regex `(foreach.*\n)(\s*\{\s*)?(.*\+\+)` in VsCode reveals a couple of more.

Reply:
Yes - you are right!